internal/tableland/impl/mesa_test.go

```diff
@@ -43,7 +43,13 @@ import (
 func TestTodoAppWorkflow(t *testing.T) {
 	t.Parallel()
 
-	ctx, tbld, backend, sc, txOpts := setup(t)
+	setup := newTablelandSetupBuilder().
```

Comment:
**Context for the refactoring**
I wanted to create a test for the relaying config. That meant adding a flag to the `setup` and `setupTablelandForTwoAddresses` procedures. Decided to refactor this ugliness into a newer approach. This is an attempt to create a better way to setup the things needed for the tests. I'm sure a lot could be improved.

**How it works?**
You have to create a builder for setting up stuff that is common to multiple `mesa` (tableland interface) instances. You do that by:

```go
setup := newTablelandSetupBuilder().
		withAllowTransactionRelay(true).
		build(t)
```

With that api, you setup things like `ethClient`, `parser`, `user store`, `system store`, and deploys the contract. You can also configure things like `AllowTransactionRelay` and `PasingOptions` with `withAllowTransactionRelay` and `withParsingOpts`.

Once you have the `setup`, you can create instances of `mesa` by calling `newTablelandClient`. 
This api helps eliminate the need for `setupTablelandForTwoAddresses`. Now i can simply call `newTablelandClient` to get another instance. For example [L237](https://github.com/tablelandnetwork/go-tableland/pull/210/files#diff-ff355983d8dcaafda2381de4717fac7f5d4bd29ddf758c1f6ba3d75e079af188R237) and  [L238](https://github.com/tablelandnetwork/go-tableland/pull/210/files#diff-ff355983d8dcaafda2381de4717fac7f5d4bd29ddf758c1f6ba3d75e079af188R238).

Reply:
Nice!